expected/12.15/type.out

```diff
@@ -995,9 +996,179 @@ SELECT * FROM "type_UUID+" WHERE "u" IS NOT NULL;
    SQLite query: SELECT `i`, coalesce(sqlite_fdw_uuid_blob(`u`),`u`), `t`, `l` FROM main."type_UUID+" WHERE ((coalesce(sqlite_fdw_uuid_blob(`u`),`u`) IS NOT NULL))
 (3 rows)
 
+--Testcase 199:
+DROP FOREIGN TABLE "type_BIT";
+--Testcase 200:
+CREATE FOREIGN TABLE "type_BIT"( "i" int OPTIONS (key 'true'), "b" bit(6)) SERVER sqlite_svr OPTIONS (table 'type_BIT');
+--Testcase 201:
+DROP FOREIGN TABLE "type_BIT+";
+ERROR:  foreign table "type_BIT+" does not exist
+--Testcase 202:
+CREATE FOREIGN TABLE "type_BIT+"( "i" int OPTIONS (key 'true'), "b" bit(6), "t" text, "l" smallint, "bi" bigint OPTIONS (column_name 'b')) SERVER sqlite_svr OPTIONS (table 'type_BIT+');
+--Testcase 203:
+INSERT INTO "type_BIT" ("i", "b") VALUES (1, 1);
+ERROR:  column "b" is of type bit but expression is of type integer
+LINE 1: INSERT INTO "type_BIT" ("i", "b") VALUES (1, 1);
+                                                     ^
+HINT:  You will need to rewrite or cast the expression.
+--Testcase 204:
+INSERT INTO "type_BIT" ("i", "b") VALUES (2, 2);
+ERROR:  column "b" is of type bit but expression is of type integer
+LINE 1: INSERT INTO "type_BIT" ("i", "b") VALUES (2, 2);
+                                                     ^
+HINT:  You will need to rewrite or cast the expression.
+--Testcase 205:
+INSERT INTO "type_BIT" ("i", "b") VALUES (3, '1');
+ERROR:  bit string length 1 does not match type bit(6)
+--Testcase 206:
+INSERT INTO "type_BIT" ("i", "b") VALUES (4, '10');
+ERROR:  bit string length 2 does not match type bit(6)
+--Testcase 207:
+INSERT INTO "type_BIT" ("i", "b") VALUES (5, '101');
+ERROR:  bit string length 3 does not match type bit(6)
+--Testcase 208:
+INSERT INTO "type_BIT" ("i", "b") VALUES (6, '110110');
+--Testcase 209:
+INSERT INTO "type_BIT" ("i", "b") VALUES (7, '111001');
+--Testcase 210:
+INSERT INTO "type_BIT" ("i", "b") VALUES (8, '110000');
+--Testcase 211:
+INSERT INTO "type_BIT" ("i", "b") VALUES (9, '100001');
+--Testcase 212:
+INSERT INTO "type_BIT" ("i", "b") VALUES (10, 53);
+ERROR:  column "b" is of type bit but expression is of type integer
+LINE 1: INSERT INTO "type_BIT" ("i", "b") VALUES (10, 53);
+                                                      ^
+HINT:  You will need to rewrite or cast the expression.
+--Testcase 213:
+SELECT * FROM "type_BIT+";
+ i |   b    |    t    | l | bi 
+---+--------+---------+---+----
+ 6 | 110110 | integer | 2 | 54
+ 7 | 111001 | integer | 2 | 57
+ 8 | 110000 | integer | 2 | 48
+ 9 | 100001 | integer | 2 | 33
+(4 rows)
+
+--Testcase 214:
+SELECT * FROM "type_BIT" WHERE b < '110110';
+ i |   b    
+---+--------
+ 8 | 110000
+ 9 | 100001
+(2 rows)
+
+--Testcase 215:
+SELECT * FROM "type_BIT" WHERE b > '110110';
+ i |   b    
+---+--------
+ 7 | 111001
+(1 row)
+
+--Testcase 216:
+SELECT * FROM "type_BIT" WHERE b = '110110';
```

Comment:
Could you add more case about bitwise operator?     
And also add EXPLAIN to see push-down ability?

Reply:
Will be added in my 2nd commit. I am studying documentation. I'll add some operations with foreign table data only and some queries with PostgreSQL constant data value. Will this normal? How do you think, @nxhai98 ?

Reply:
Could you give me an example? I do not understand as well.

Reply:
See in https://github.com/pgspider/sqlite_fdw/pull/83/commits/9ec823394aa94b51c14a0e7f446e33ffa1a95b39, 
```sql
SELECT b1."i" "i₁", b1."b" "b₁", b2."i" "i₂", b2."b" "b₂", b1."b" | b2."b" "res" FROM "type_BIT" b1 INNER JOIN "type_BIT" b2 ON true;
SELECT b1."i" "i₁", b1."b" "b₁", b2."i" "i₂", b2."b" "b₂", b1."b" & b2."b" "res" FROM "type_BIT" b1 INNER JOIN "type_BIT" b2 ON true;
SELECT b1."i" "i₁", b1."b" "b₁", b2."i" "i₂", b2."b" "b₂", b1."b" # b2."b" "res" FROM "type_BIT" b1 INNER JOIN "type_BIT" b2 ON true;
```
Unfortunately there is no pushdowning for listed operators **`|`** **`&`** **`#`**. Could you help me with name of function in `deparse.c` where I can add pushdowning?

Reply:
You mean to using bitwise operator with a Postgres constant like: `b2."b" & B'1001'`?   
If so, there is no problem.    

> Unfortunately there is no pushdowning for listed operators | & #. Could you help me with name of function in deparse.c where I can add pushdowning?

`sqlite_fdw` already support push-down bitwise operator.    
However, almost operators/functions cannot push down in SELECT clause (similar with your TCs) as PostgreSQL spec (I'm not sure about the reason).    
bitwise operators can push down in WHERE clause (some other clause: HAVING, ORDER BY,...) normally:
```sql
--Testcase 261:
EXPLAIN VERBOSE
SELECT "i", "b", "b" >> 2 "res" FROM "type_VARBIT" WHERE ("b" >> 2) > ("b" & "b");
                                         QUERY PLAN                                         
--------------------------------------------------------------------------------------------
 Foreign Scan on public."type_VARBIT"  (cost=10.00..608.52 rows=607 width=53)
   Output: i, b, ((b)::"bit" >> 2)
   SQLite query: SELECT `i`, `b` FROM main."type_VARBIT" WHERE (((`b` >> 2) > (`b` & `b`)))
(3 rows)
```

Reply:
> You mean to using bitwise operator with a Postgres constant like: `b2."b" & B'1001'`?

Yes. Some of this TCs added after TC 264. See https://github.com/pgspider/sqlite_fdw/pull/83/commits/77b0efb62db35bea88b8752d43ad6266bbcfcb00.

> bitwise operators can push down in WHERE clause (some other clause: HAVING, ORDER BY,...) normally

Added TCs for `WHERE` case.

I have noted `deparse.c` deny **`~`** operator and change this behaviour. Because there is no **`#`** operator in SQLite, this was be deny in new `deparse.c` code.

Reply:
Thank you for updating. Please help me to resolve this thread.   
About push-down ability, I added some more comments, please help me to check it.